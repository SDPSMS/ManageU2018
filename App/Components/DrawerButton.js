import React from 'react'
import Icon from 'react-native-vector-icons/MaterialIcons'

/**
 * Button to toggle open/close the drawer.
 * @param navigation
 * @returns {*}
 */
// TODO: Change with button. (Nav drawer button.)
const drawerButton = (navigation) => {
  // TODO: Needs redux implementation to check the type.
  // the navigation function has to be tested.
  return (
    <Icon
      name={'menu'}
      size={20}
      onPress={() => { navigation.state.index === 0 ? navigation.navigate('DrawerOpen') : navigation.navigate('DrawerClose') }}
      style={{marginLeft: 10, color: 'white'}}
    />
  )
}

export default drawerButton
