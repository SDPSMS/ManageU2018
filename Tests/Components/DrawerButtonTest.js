import 'react-native'
import React from 'react'
import DrawerButton from '../../App/components/DrawerButton'

import renderer from 'react-test-renderer'

test('Drawer button renders correcty', () => {
  const tree = renderer.create(<DrawerButton />).toJSON()
  expect(tree).toMatchSnapshot()
})
