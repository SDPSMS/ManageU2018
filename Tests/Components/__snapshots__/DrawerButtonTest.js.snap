// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer button renders correcty 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 20,
      },
      Object {
        "color": "white",
        "marginLeft": 10,
      },
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  î—’
</Text>
`;
